### PR TITLE
Make fullscreen mode as a default state for Android devices

### PR DESCRIPTION
**Pull request by ihhub**
To have better initial experience with the game.

relates to #6008

### CODE REVIEW

File: src/fheroes2/system/settings.cpp [hunk 1]
```diff
@@ -114,6 +114,11 @@ Settings::Settings()
     _optGlobal.SetModes( GLOBAL_BATTLE_SHOW_MOVE_SHADOW );
     _optGlobal.SetModes( GLOBAL_BATTLE_AUTO_SPELLCAST );
 
+#if defined( ANDROID )
+    // Due to the nature of Android devices having small screens in general it is good to make fullscreen option by default.
+    _optGlobal.SetModes( GLOBAL_FULLSCREEN );
+#endif
```
Comment (oleg-derevenetz) [inline src/fheroes2/system/settings.cpp:120]: Hi @ihhub what do you think about moving the device-dependent logic to the `engine`? Say, add something like `isHandheldDevice()` to the `engine` and use it here?

Comment (ihhub) [reply src/fheroes2/system/settings.cpp:120]: Yup, we can do it.

File: src/fheroes2/system/settings.cpp [hunk 2]
```diff
@@ -114,10 +115,10 @@ Settings::Settings()
     _optGlobal.SetModes( GLOBAL_BATTLE_SHOW_MOVE_SHADOW );
     _optGlobal.SetModes( GLOBAL_BATTLE_AUTO_SPELLCAST );
 
-#if defined( ANDROID )
-    // Due to the nature of Android devices having small screens in general it is good to make fullscreen option by default.
-    _optGlobal.SetModes( GLOBAL_FULLSCREEN );
-#endif
+    if ( fheroes2::isHandheldDevice() ) {
+        // Due to the nature of Android devices having small screens in general it is good to make fullscreen option by default.
```
Comment (oleg-derevenetz) [inline src/fheroes2/system/settings.cpp]:
```suggestion
        // Due to the nature of handheld devices having small screens in general it is good to make fullscreen option by default.
```

Comment (ihhub) [reply src/fheroes2/system/settings.cpp]: It is in the code🙂

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**